sysl2/sysl/ints.go

```diff
@@ -0,0 +1,253 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/anz-bank/sysl/src/proto"
+	log "github.com/sirupsen/logrus"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+type IntegrationParam struct {
+	m            *sysl.Module
+	excludes     StrSet
+	passthroughs StrSet
+	commonEndpts StrSet
+	highlights   StrSet
+	apps         StrSet
+	deps         map[string]*AppDependency
+}
+
+type ApplicationCollectionElement struct {
+	apps StrSet
+}
+
+type IntegrationStatementElement struct {
+	appname string
+	epname  string
+	stmts   []*sysl.Statement
+}
+
+func MakeIntegrationParam(mod *sysl.Module, excludes, passthroughs, highlights StrSet) *IntegrationParam {
+	return &IntegrationParam{
+		m:            mod,
+		excludes:     excludes,
+		passthroughs: passthroughs,
+		commonEndpts: MakeStrSet(".. * <- *", "*"),
+		highlights:   highlights,
+		apps:         StrSet{},
+		deps:         map[string]*AppDependency{},
+	}
+}
+
+func MakeApplicationCollectionElement(apps StrSet) *ApplicationCollectionElement {
+	return &ApplicationCollectionElement{apps}
+}
+
+func MakeIntegrationStatementElement(appname, epname string, stmts []*sysl.Statement) *IntegrationStatementElement {
+	return &IntegrationStatementElement{appname, epname, stmts}
+}
+
+func (v *IntegrationParam) collectApplicationDependencies(e *ApplicationCollectionElement) {
+	for app := range e.apps {
+		for epname, endpt := range v.m.GetApps()[app].GetEndpoints() {
+			i := MakeIntegrationStatementElement(app, epname, endpt.GetStmt())
+			v.handleStatement(i)
+		}
+	}
+}
+
+func (v *IntegrationParam) handleStatement(i *IntegrationStatementElement) {
```

Comment:
`handleStatement` should only need `[]*sysl.Statement`

Reply:
The same reason as the one up.

Reply:
yep. I'll follow this advice.